### PR TITLE
dxf: systable and scheduler change for modify task at runtime

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: ref #57497

Problem Summary:

### What changed and how does it work?
Add a new task state `modifying` and add a new field `modify_params` which holds parameters about current modification. the system table change part has verified with PM @frank945946
Add some basic interface in storage for modify and add a empty impl in scheduler

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)

upgrade from 8.1, success, task and history table matches
bootstrap with master, success, task and history table matches

- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by joccau — APPROVED** [review]
LGMT

File: pkg/session/bootstrap.go [hunk 1]
```diff
@@ -1375,6 +1381,7 @@ var (
 		upgradeToVer216,
 		upgradeToVer217,
 		upgradeToVer218,
+		upgradeToVer239,
```
Comment (fzzf678) [inline pkg/session/bootstrap.go:1384]: Please consult that why the version is changed from 218 to 239, not 219?

**Review by fzzf678 — APPROVED** [review]
rest LGTM

File: pkg/session/bootstrap.go (hunk 1 above)
Comment (D3Hunter) [reply pkg/session/bootstrap.go:1384]: see the comment
```
	// ...
	// [version219, version238] is the version range reserved for patches of 8.5.x
	// ...

	// next version should start with 239
```

**Review by YuJuncen — APPROVED** [review]

**Review by yudongusa — APPROVED** [review]

**Comment by D3Hunter** [comment]
/approve

for disttask pkg

File: pkg/disttask/framework/proto/modify.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+type ModifyParam struct {
+	PrevState     TaskState      `json:"prev_state"`
+	Modifications []Modification `json:"modifications"`
+}
+
+// Modification is one modification for task.
+type Modification struct {
+	Type ModificationType `json:"type"`
+	To   int64            `json:"to"`
+}
```
Comment (tangenta) [inline pkg/disttask/framework/proto/modify.go:40]: If we add a new `ModificationType` in the future like `ModifyServiceScope`, how to change this struct?

File: pkg/session/bootstrap.go [hunk 3]
```diff
@@ -611,8 +612,9 @@ const (
 		step INT(11),
 		target_scope VARCHAR(256) DEFAULT "",
 		error BLOB,
+		modify_params json,
```
Comment (tangenta) [inline pkg/session/bootstrap.go:615]: Is this column necessary for history table?

Comment (D3Hunter) [reply pkg/session/bootstrap.go:615]: just make sure they have the same table structure, we have a UT for this, and when transfering to history, we use `insert into ... select *`

File: pkg/disttask/framework/proto/modify.go (hunk 2 above)
Comment (D3Hunter) [reply pkg/disttask/framework/proto/modify.go:40]: just add another `ModificationType` enum, we can add a `ToStr` for this case

**Pull request merged**